en/guide/message_signing.md

```diff
@@ -1,121 +1,144 @@
-{
-	// Make the assumption that channel 0 is USB and should always be accessible
-	if (status == mavlink_get_channel_status(MAVLINK_COMM_0)) {
-		return true;
-	}
+When a signed packet arrives it should be discarded if the:
+* Timestamp is older than the previous packet from the same logical stream - where a logical stream is defined as the sequence of MAVLink packets with the same (`SystemID`, `ComponentID`, `LinkID`) tuple.
+* Computed 48 bit signature does not match the signature included in the packet. 
+* The timestamp is more than 1 minute (6,000,000) behind the local system’s timestamp.
 
-	for (unsigned i = 0; i < sizeof(unsigned_messages) / sizeof(unsigned_messages[0]); i++) {
-		if (unsigned_messages[i] == message_id) {
-			return true;
-		}
-	}
 
-	return false;
-}
-```
+## Accepting Unsigned Packets
 
-## Handling Link IDs
+The rules for accepting *unsigned* packets will be implementation specific, but could be based on a combination of a parameter setting, transport type, message type, (in)compatibility flags etc.
 
-The purpose of the `link_id` field in the *MAVLink 2* signing structure is to prevent cross-channel replay attacks. Without the `link_id` an attacker could record a packet (such as a disarm request) on one channel, then play it back on a different channel.
+> **Note** MAVLink libraries should provide a mechanism that allows an implementation to make this kind of decision.
 
-The intention with the link IDs is that each channel of communication between an autopilot and a GCS uses a different link ID. There is no requirement that the same link ID be used in both directions however.
+A few rules that may be useful for some systems are:
+* All unsigned packets accepted based on a system-specific parameter.
+* All unsigned packets accepted if the connection is over a "secure channel" (e.g. local USB cable or local wired Ethernet cable).
+* `RADIO_STATUS` packets are always accepted without signing (to make life easier for telemetry radios).
+* All unsigned packets accepted when in an "unsigned mode" (perhaps triggered by a hardware button pressed on boot).
+* MAVLink 1 messages may always be accepted or always rejected, depending on the system requirements.
+* All other packets will be rejected if not signed!
 
-For C implementations the obvious mechanism is to use the MAVLink channel number as the link ID. That works well for an autopilot, but runs into an issue for a GCS implementation. The issue is that a user may launch multiple GCS instances talking to the same autopilot via different communication links (such as two radios, or USB and a radio). These multiple GCS instances will not be aware of each other, and so may choose the same link ID. If that happens then a large number of correctly signed packets will be rejected by the autopilot as they will have timestamps that are older than the timestamp received for the same stream tuple on the other communication link.
+<!-- 
```

Comment:
The proposal doc talks about handling the case where a signed packet has an incorrect signature/timestamp but would be accepted if unsigned. E.g. for lost copter finding.

Was anything agreed in standard for this? Was anything done in the C library?